File: WeekendWarriors/Questionnaires/Questionnaires/QLS/AST/Section.cs
using System.Collections.Generic;

namespace Questionnaires.QLS.AST
{
    public class Section : INode
    {
        public string Name {  get; }
        public List<Section> Sections { get; }
        public List<Question> Questions { get; }
        public List<DefaultStyle> Styles { get; }
        public Section(string name)
        {
            Sections = new List<Section>();
            Questions = new List<Question>();
            Styles = new List<DefaultStyle>();
            this.Name = name;
        }

        public void AddChild(Section section)
        {
            Sections.Add(section);
        }

        public void AddChild(Question question)
        {
            Questions.Add(question);
        }

        public void AddChild(DefaultStyle style)
        {
            Styles.Add(style);
        }

    }
}

File: Loading/QLS2/src/qls/ast/Section.java
package qls.ast;

import java.util.List;

import QL.ast.Node;

public class Section extends Node {
	
	private final String name;
	private final List<Question> questions;

	public Section(String name, List<Question> questions, int line) {
		super(line);
		this.name = name;
		this.questions = questions;
	}

}
